adaptdl/adaptdl/torch/parallel.py

```diff
@@ -160,11 +173,11 @@ def to_tensorboard(self, writer, global_step, tag_prefix=""):
         if tag_prefix and not tag_prefix.endswith("/"):
             tag_prefix += "/"
         writer.add_scalar(tag_prefix + "Gradient_Norm_Sqr",
-                          self.adascale.sqr_avg(), global_step)
+                          self.gns.sqr_avg(), global_step)
         writer.add_scalar(tag_prefix + "Gradient_Variance",
-                          self.adascale.var_avg(), global_step)
+                          self.gns.var_avg(), global_step)
         writer.add_scalar(tag_prefix + "Learning_Rate_Factor",
-                          self._state.gain, global_step)
+                          self._state.lr_factor, global_step)
```

Comment:
Is this the only place where `lr_factor` is needed from state and isn't recomputed?

Reply:
If, so I think we can just recompute it here as well. If it needs to be fresh each time, then I don't think there is much point in having it in state.

Reply:
nvm, needs the current scale, which requires state